grammar/expr.lyg

```diff
@@ -125,6 +127,6 @@ ElseExpr =
   | If:If
   ;
 
-MatchArm = attrs:OuterAttr* "|"? pats:Pat+ % "|" { "if" guard:Expr }? "=>" body:Expr ","?;
+MatchArm = attrs:OuterAttr* "|"? pat:Pat { "if" guard:Expr }? "=>" body:Expr ","?;
```

Comment:
According to this setup, where nested or-patterns are unstable, then `match 0 { 0 | 1 => 0 }` is also unstable but it isn't.

Reply:
We don't have a proper stability setup, but I can undo this change.

Reply:
Maybe just leave a comment instead. You might want to double check `if/while let` also.

Reply:
Also, we annoyingly don't allow `fn foo(Ok(x) | Err(x): _);` but require parens around the pattern instead.